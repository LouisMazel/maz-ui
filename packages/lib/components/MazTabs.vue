<template>
  <div class="m-tabs">
    <slot></slot>
  </div>
</template>

<script lang="ts" setup>
  import { computed, provide, ref, type Ref } from 'vue'

  const props = defineProps<{
    /** The the selected tab number */
    modelValue?: number
  }>()

  const emits = defineEmits<{
    /**
     * Emitted when the selected tab change
     * @property {number} newValue new value set
     */
    (event: 'update:model-value', index: number): void
  }>()

  const localValue = ref(1)

  const currentTab = computed({
    get: () => props.modelValue ?? localValue.value,
    set: (index: number) => {
      localValue.value = index
      emits('update:model-value', index)
    },
  })

  function updateCurrentTab(index: number) {
    currentTab.value = index

    return index
  }

  export type MazTabsProvide = {
    currentTab: Ref<number>
    updateCurrentTab: (index: number) => number
  }

  provide<MazTabsProvide>('maz-tabs', {
    currentTab,
    updateCurrentTab,
  })
</script>
