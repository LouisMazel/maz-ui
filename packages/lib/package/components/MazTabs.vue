<script lang="ts" setup>
  // import { provide, inject } from 'vue'
</script>
