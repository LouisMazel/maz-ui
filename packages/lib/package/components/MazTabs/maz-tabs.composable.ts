import { ref } from 'vue'

const currentTab = ref<number>()

export const useMazTabs = () => {
  return {
    currentTab,
  }
}
oo
