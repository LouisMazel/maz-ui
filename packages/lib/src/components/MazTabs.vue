<script lang="ts" setup>
import type { Ref } from 'vue'
import { computed, provide, ref } from 'vue'

export interface MazTabsProps {
  /** The the selected tab number */
  modelValue?: number
}

const props = defineProps<MazTabsProps>()

const emits = defineEmits<{
  /**
   * Emitted when the selected tab change
   * @property {number} newValue new value set
   */
  'update:model-value': [value: number]
}>()

const localValue = ref(1)

const currentTab = computed({
  get: () => props.modelValue ?? localValue.value,
  set: (index: number) => {
    localValue.value = index
    emits('update:model-value', index)
  },
})

function updateCurrentTab(index: number) {
  currentTab.value = index

  return index
}

export interface MazTabsProvide {
  currentTab: Ref<number>
  updateCurrentTab: (index: number) => number
}

provide<MazTabsProvide>('maz-tabs', {
  currentTab,
  updateCurrentTab,
})
</script>

<template>
  <div class="m-tabs m-reset-css">
    <slot />
  </div>
</template>
