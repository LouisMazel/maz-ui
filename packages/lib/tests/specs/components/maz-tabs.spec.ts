import MazTabs from '@components/MazTabs.vue'
import { mount } from '@vue/test-utils'

describe('mazTabs.vue', () => {
  it('should display the content', () => {
    const wrapper = mount(MazTabs, {
      slots: {
        default: '<div>Contenu du slot</div>',
      },
    })

    expect(wrapper.text()).toContain('Contenu du slot')
  })

  it('initialize component with good value', () => {
    const wrapper = mount(MazTabs)

    expect(wrapper.vm.currentTab).toBe(1)
  })
})
