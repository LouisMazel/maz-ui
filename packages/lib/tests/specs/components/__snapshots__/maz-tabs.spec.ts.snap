// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MazTabs.vue > Should match with the snapshot 1`] = `"<div class=\\"m-tabs\\"></div>"`;
